File: endpoints/src/main/java/com/cromoteca/samples/endpoints/AccessorsEndpoint.java
package com.cromoteca.samples.endpoints;

import com.vaadin.flow.server.auth.AnonymousAllowed;
import dev.hilla.Endpoint;

@Endpoint
@AnonymousAllowed
public class AccessorsEndpoint {
  public void accessors(Accessors arg) {}

  public static class Accessors {
    public int propertyWithField;

    private int propertyWithFieldAndGetter;

    public int getPropertyWithFieldAndGetter() {
      return propertyWithFieldAndGetter;
    }

    private int propertyWithFieldAndSetter;

    public void setPropertyWithFieldAndSetter(int propertyWithFieldAndSetter) {
      this.propertyWithFieldAndSetter = propertyWithFieldAndSetter;
    }

    private int propertyWithFieldAndGetterAndSetter;

    public int getPropertyWithFieldAndGetterAndSetter() {
      return propertyWithFieldAndGetterAndSetter;
    }

    public void setPropertyWithFieldAndGetterAndSetter(int propertyWithFieldAndGetterAndSetter) {
      this.propertyWithFieldAndGetterAndSetter = propertyWithFieldAndGetterAndSetter;
    }

    public int getPropertyWithGetter() {
      return 0;
    }

    public void setPropertyWithSetter(int arg) {}

    public int getPropertyWithGetterAndSetter() {
      return 0;
    }

    public void setPropertyWithGetterAndSetter(int arg) {}
  }
}

File: generator/src/test/resources/com/cromoteca/samples/endpoints/AccessorsEndpoint.ts
import client from '../../../../connect-client.default.js';
import type Accessors from './AccessorsEndpoint/Accessors.js';
import { type EndpointRequestInit } from '@hilla/frontend';

async function accessors(arg: Accessors, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.samples.endpoints.AccessorsEndpoint', 'accessors', { arg }, init);
}

const AccessorsEndpoint = {
    accessors,
};

export default AccessorsEndpoint;
